execution/transactor.go

```diff
@@ -224,17 +185,17 @@ func (trans *transactor) Transact(privKey []byte, address *acm.Address, data []b
 	}
 
 	// Got ourselves a tx.
-	txS, errS := trans.SignTx(tx, []acm.PrivateAccount{pa})
-	if errS != nil {
-		return nil, errS
+	err = tx.Sign(trans.tip.ChainID(), inputAccount)
```

Comment:
Ah, so we integrate ChainID here in the signer, that's why you've moved it to tip, no?